packages/utils/regl-renderer/src/geometry-utils-V2/geom3ToGeometries.js

```diff
@@ -10,128 +10,134 @@ const {toArray} = require('../utils')
-    normalPositionLookup = {}
-    let tupplesIndex = 0
+  const polygons = geometry.polygons
 
-    for (let i = 0; i < polygons.length; i++) {
-      const polygon = polygons[i]
+  let normalPositionLookup = []
+  normalPositionLookup = {}
+  let tupplesIndex = 0
 
-      const color = polygonColor(polygon, faceColorRgb)
-      const rawNormal = polygon.plane
-      const normal = [rawNormal[0], rawNormal[1], rawNormal[2]]
+  // FIXME this isn't going to work if the polygons are > 65000 (see below)
 
-      if (color[3] !== 1) {
-        isTransparent = true
-      }
+  for (let i = 0; i < polygons.length; i++) {
+    const polygon = polygons[i]
 
-      const polygonIndices = []
-      // we need unique tupples of normal + position , that gives us a specific index (indices)
-      // if the angle between a given normal and another normal is less than X they are considered the same
-      for (let j = 0; j < polygon.vertices.length; j++) {
-        let index
-
-        const vertex = polygon.vertices[j]
-        const position = [vertex[0], vertex[1], vertex[2]]
-
-        if (smoothLighting) {
-          const candidateTupple = {normal, position}
-          const existingTupple = fuzyNormalAndPositionLookup(normalPositionLookup, candidateTupple, normalThreshold)
-          if (!existingTupple) {
-            const existingPositing = normalPositionLookup[candidateTupple.position]
-            const itemToAdd = [{normal: candidateTupple.normal, index: tupplesIndex}]
-            if (!existingPositing) {
-              normalPositionLookup[candidateTupple.position] = itemToAdd
-            } else {
-              normalPositionLookup[candidateTupple.position] = normalPositionLookup[candidateTupple.position]
-                .concat(itemToAdd)
-            }
-            index = tupplesIndex
-            // normalPositionLookup.push(candidateTupple)
-            // index = normalPositionLookup.length - 1
-            if (faceColor !== undefined) {
-              colors.push(color)
-            }
-            normals.push(normal)
-            positions.push(position)
-            tupplesIndex += 1
+    const faceColor = polygonColor(polygon, color)
+    const rawNormal = polygon.plane
+    const normal = [rawNormal[0], rawNormal[1], rawNormal[2]]
+
+    if (faceColor && faceColor[3] !== 1) {
+      isTransparent = true
+    }
+
+    const polygonIndices = []
+    // we need unique tupples of normal + position , that gives us a specific index (indices)
+    // if the angle between a given normal and another normal is less than X they are considered the same
+    for (let j = 0; j < polygon.vertices.length; j++) {
+      let index
+
+      const vertex = polygon.vertices[j]
+      const position = [vertex[0], vertex[1], vertex[2]]
+
+      if (options.smoothLighting) {
+        const candidateTupple = {normal, position}
+        const existingTupple = fuzyNormalAndPositionLookup(normalPositionLookup, candidateTupple, options.normalThreshold)
```

Comment:
why use the one from options ? we already have it unpacked, and the correct mix of default and user defined value ?

Reply:
Ah. I see... I reorganized the code slightly, breaking out the convert() function. The options are used directly in this function.

```
const convert = (options, geometry) => {
...
}
```